app/src/main/java/com/andrew/tasky/agenda/presentation/screens/agenda/AgendaViewModel.kt

```diff
@@ -130,6 +143,17 @@ class AgendaViewModel@Inject constructor(
         }
     }
 
+    private val logoutResultChannel = Channel<AuthResult<Unit>>()
+    val logoutResults = logoutResultChannel.receiveAsFlow()
+
+    fun logout() {
+        viewModelScope.launch {
+            withContext(NonCancellable) {
+                logoutResultChannel.send(authRepository.logout())
```

Comment:
Is the result here really necessary? What would you do if logging out fails, in case the user is offline for example? They'd still want to logout, so do you really care about an error?